netket/graph/grid.py

```diff
@@ -122,28 +186,128 @@ def __repr__(self):
             pbc = self.pbc
         return f"Grid(length={self.length}, pbc={pbc})"
 
-    def periodic_translations(self) -> List[List[int]]:
+    def translations(self, dim: int = None, period: int = 1) -> SymmGroup:
         """
         Returns all permutations of lattice sites that correspond to translations
         along the grid directions with periodic boundary conditions.
 
         The periodic translations are a subset of the permutations returned by
         `self.automorphisms()`.
+
+        Arguments:
+            dim: If set, only translations along `dim` will be returned.
+            period: Period of the translations; should be a divisor of the length in
+                the corresponding lattice dimension.
```

Comment:
```suggestion
            period: Period of the translations; should be a divisor of the length in
                the corresponding lattice dimension. ( For example, in a PBC chain of length 8, 
                period=1 will return all 8 translations, period=2 with return 4 translations, etc...)
```

Reply:
I've noticed that `period` is a misleading name so I've changed it to `step`. Should be clearer now.

Reply:
I agree 100%